### PR TITLE
Allowed title as props for `<details>`

### PR DESCRIPTION
**Pull request by yathomasi**
From #3383 
> Additionally, should we use a `title` prop for the anchor base instead of the first child's text?

Used title as props for `<details>` tag. 

Initially, I only used `title` as props but there are 2-3 particular cases that use `code` highlighting in the title. 
Screen Shot 2022-03-24 at 19 12 52
So, I also added back the first child's text as a fallback if no title is passed as props so that we still can use markdown as well.
Also, previously in this particular case, the code section seems to be parsed as `object object` as well 
resulting: https://dvc.org/doc/user-guide/external-dependencies#expand-to-see-resulting-object-object-file
This issue should also be fixed with the pr.

But, if markdown syntax is not so important and the title would be enough then the code implementation of the Details component will be very simple and preferable.

### CODE REVIEW

**Comment by yathomasi** [comment]
> question if this approach breaks some titles and we have to maintain two options, why do we consider it superior and why do? let's keep the previous one?

markdown wouldn't be supported when passed as props but I think it would be okay without it too as it's used in just 2 titles to highlight `.dvc`. 
And, using `title` as props should make consistent with other markdown custom elements as they also support through props.

> Also, previously in this particular case, the code section seems to be parsed as `object object` as well
> resulting: https://dvc.org/doc/user-guide/external-dependencies#expand-to-see-resulting-object-object-file
> This issue should also be fixed with the pr.

I think this one needs to be addressed and which we didn't address on pervious pr.

**Review by rogermparent — COMMENTED** [review]
Not a bad idea, but I think it would be better to separate the bugfixes like the `object object` thing and the prop-title feature into different PRs, the former is an easy fix we'll want asap while the latter needs some more discussion.

File: content/docs/command-reference/checkout.md [hunk 1]
```diff
@@ -112,9 +112,7 @@ pipeline stages, such as the <abbr>DVC project</abbr> created for the
 [Get Started](/doc/start). Then we can see what happens with `git checkout` and
 `dvc checkout` as we switch from tag to tag.
 
-<details>
-
-### Click and expand to set up the project
+<details title="Click and expand to set up the project>
```
Comment (jorgeorpinel) [inline content/docs/command-reference/checkout.md:115]: Does this PR update all existing `<details>` blocks? Thanks

**Comment by jorgeorpinel** [comment]
Ah shoot, due to #3391 there are some merge conflicts to solve now... ⌛

**Comment by shcheklein** [comment]
Reminder:

> Still, I don't see any clear motivation for this tbh. And still it breaks things in some cases. So, my take - let's postpone / keep it as-is for now.

can we skip addressing the conflict? 

please please let's be mindful about the bigger priorities

File: content/docs/command-reference/import-url.md [hunk 2]
```diff
@@ -168,9 +168,7 @@ produces a regular stage in `dvc.yaml`.
 To illustrate these examples we will be using the <abbr>project</abbr> explained
 in the [Get Started](/doc/start).
 
-<details>
-
-### Click and expand to set up example
+<details title="Click and expand to set up example">
```
Comment (jorgeorpinel) [inline content/docs/command-reference/import-url.md:171]: OK just noticed we're postponing this. Just a note that per #3391 these titles should no longer start in "Click for" nor end in "example". Thanks

**Comment by jorgeorpinel** [comment]
Given this is not much of a priority + all the conflicts, should we close it or turn it into an issue? Thanks

**Pull request closed without merging**